pkg/resources/deployments/deployment.go

```diff
@@ -46,7 +145,14 @@ func NewForCluster(mi *miniov1beta1.MirrorInstance) *appsv1.Deployment {
 		},
 		Spec: appsv1.DeploymentSpec{
 			Replicas: &replicas,
-			Template: corev1.PodTemplateSpec{},
+			Selector: mi.Spec.Mcs.Selector,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: mcsMetadata(mi),
+				Spec: corev1.PodSpec{
+					Containers:    []corev1.Container{mcsContainer(mi, serviceName)},
+					RestartPolicy: "Always",
```

Comment:
On-Failure

Reply:
mcs is not allowed to exit otherwise, if it does it means there is a bug in mcs lets not mask it.

Reply:
On "Always" restart policy is supported by deployments @harshavardhana 
https://github.com/kubernetes/kubernetes/issues/24725